### PR TITLE
Adding cody ignore docs

### PR DESCRIPTION
**Pull request by arafatkatze**
Adding temporary docs for cody ignore(Mostly copying the instructions @abeatrix posted already)

### CODE REVIEW

File: vscode/doc/codyignore.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Cody Ignore (Experimental feature)
```
Comment (abeatrix) [inline vscode/doc/codyignore.md:1]: I would emphasize on this is an experimental feature for internal dogfooding only as it doesn't work with local embeddings yet.
```suggestion
## Cody Ignore (Internal Experimental)
```

File: vscode/doc/codyignore.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Cody Ignore is an experimental feature that functions similarly to `.gitignore`, but it's specifically designed for the context consumed by Cody, the code assistant.
```
Comment (abeatrix) [inline vscode/doc/codyignore.md:3]:
```suggestion
Cody Ignore is an experimental feature currently available for internal testing. It functions similarly to `.gitignore`, but is specifically designed for the context consumed by Cody, the code assistant. It enables users to specify files they don't want to send to a third party LLM providers.
```

File: vscode/doc/codyignore.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+2. In the root of your workspace, create a `.cody` directory.
+3. Inside the `.cody` directory, create a file named `ignore`.
+4. Edit the `.cody/ignore` file to include patterns for the files and directories you want to ignore. For example, to ignore everything in a directory named `src`, you would add the following line: 
+src/
+5. To test if it's working, open a file from the directory you've chosen to ignore and highlight some code.
+6. Ask Cody a question about the highlighted code.
+7. Check the file context in Cody's response. You should not see any files from the ignored directory being used as context.
+8. Note that autocomplete features will also not work on files specified in the `.cody/ignore` file.
+
+Remember, this is an experimental feature, so use it with caution and report any issues you encounter to help improve the functionality.
```
Comment (abeatrix) [inline vscode/doc/codyignore.md:23]:
```suggestion
Remember, this is an experimental feature for internal testing only, so use it with caution and report any issues you encounter to help improve the functionality.
```

**Review by abeatrix — APPROVED** [review]
Perfect! I think this is a good start for internal doc too. Thanks for adding this.

I just created a PR to clean up cody ignore too, and set it up in the Cody repo for testing purpose, would appreciate your review https://github.com/sourcegraph/cody/pull/2913

Thanks again for adding this!

Made a few inline suggestions to emphasize that this is for internal testing only, and point out it's for preventing files from being sent to third party llm providers! Feel free to ignore them if you disagree.

**Comment by arafatkatze** [comment]
Thanks @abeatrix

**Pull request merged**